Sources/Paywalls/Components/PaywallLinkButtonComponent.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  PaywallLinkButtonComponent.swift
+//
+//
+//  Created by James Borthwick on 2024-08-21.
+//
+// swiftlint:disable missing_docs
+
+import Foundation
+
+#if PAYWALL_COMPONENTS
+
+public extension PaywallComponent {
+
+    struct LinkButtonComponent: PaywallComponentBase {
+
+        let type: String
+        public let url: URL
+        public let textComponent: PaywallComponent.TextComponent
+
+        public init(
+            url: URL,
+            textComponent: PaywallComponent.TextComponent
+        ) {
+            self.type = "link_button"
```

Comment:
We can definitely improve this and write it as:

`let type: ComponentType = .text`

Thanks!

Reply:
Xcode would rather we set it in the initializers, so that the decoder initializer can set it. But really it should never be anything but a hardcoded value....

<img width="794" alt="image" src="https://github.com/user-attachments/assets/b5a07976-c808-4259-989a-fc570463e16e">